src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -475,6 +475,45 @@ SHAMap::firstBelow(
     }
     return nullptr;
 }
+SHAMapLeafNode*
+SHAMap::lastBelow(
+    std::shared_ptr<SHAMapTreeNode> node,
+    SharedPtrNodeStack& stack,
+    int branch) const
+{
+    // Return the first item at or below this node
+    if (node->isLeaf())
+    {
+        auto n = std::static_pointer_cast<SHAMapLeafNode>(node);
+        stack.push({node, {leafDepth, n->peekItem()->key()}});
+        return n.get();
+    }
+    auto inner = std::static_pointer_cast<SHAMapInnerNode>(node);
+    if (stack.empty())
+        stack.push({inner, SHAMapNodeID{}});
+    else
+        stack.push({inner, stack.top().second.getChildNodeID(branch)});
+    for (int i = branchFactor; i >= 0;)
```

Comment:
This looks like an off by one error. `branchFactor` isn't a valid index. I think we want to start at `branchFactor-1`

Reply:
good catch